### PR TITLE
Refactored site-title to be a proper redux middleware

### PR DESCRIPTION
**Pull request by yurynix**
Currently `site-title` monkey patching redux store's `dispatch` method and looses
`dispatch` result.

That PR solves that by making it a proper redux middleware.

#### Testing instructions
  
1. Run `git checkout update/screen-title-middleware` and start your server, or open a [live branch](https://calypso.live/?branch=update/screen-title-middleware)
2. Go to page that uses `DocumentHead` element (redux state document titles), for example the [edit post page page](http://calypso.localhost:3000/post/)
3. Titles should work and vary `New` for a new post, `Edit Post` for existing post edit.
  
#### Reviews
  
- [x] Code
- [x] Product
- [x] Tests
   
@Automattic/sdev-feed

### CODE REVIEW

File: client/lib/screen-title/index.js [hunk 1]
```diff
@@ -19,26 +19,24 @@ const TITLE_UPDATING_ACTIONS = [
 ];
 
 /**
- * Given a Redux store instance, subscribes to the store, updating the page
+ * A middleware that updating the page
```
Comment (drewblaisdell) [inline client/lib/screen-title/index.js]: Super minor grammar note, but this should be `Redux middleware that updates the page title` or something.

File: client/lib/screen-title/index.js [hunk 2]
```diff
@@ -19,26 +19,24 @@ const TITLE_UPDATING_ACTIONS = [
...
  * Flux actions used elsewhere in the codebase.
  *
  * @param {Object} store Redux store instance
+ * @returns {Function} A configured middleware with store
  */
-export function subscribeToStore( store ) {
-	const dispatch = store.dispatch;
-	store.dispatch = function( action ) {
-		dispatch( ...arguments );
+export const screenTitleMiddleware = store => next => action => {
```
Comment (drewblaisdell) [inline client/lib/screen-title/index.js]: - We might want to update the README to indicate that this is middleware.
- We might want to move this to `lib/screen-title-middleware` to match the name of the function.

**Comment by drewblaisdell** [comment]
Thanks for fixing this. The code looks good to me, and I have asserted that this updates the page title. I added a commit to update the README.

**Comment by spncrb** [comment]
Works as described. 👍

**Comment by yurynix** [comment]
Thanks @drewblaisdell and @spncrb

**Pull request merged**